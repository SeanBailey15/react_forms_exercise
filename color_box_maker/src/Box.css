.Box {
  display: flex;
  margin-bottom: 10px;
}

.Box-div {
  margin-left: 10px;
  border: 1px solid black;
}

.Box-btn {
  font-weight: bolder;
  border: 3px solid black;
  border-radius: 10px;
  width: 30px;
  height: 30px;
}
