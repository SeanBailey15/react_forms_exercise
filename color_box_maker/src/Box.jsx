import React from "react";
import './Box.css'

const Box = ({id, width, height, color, removeBox}) => {
    const handleRemove = () => {
        removeBox(id)
    };

    return (
        <div className="Box">
            <button className="Box-btn" type="submit" onClick={handleRemove} style={{display: "inline-block"}}>X</button>
            <div className="Box-div" data-testid="box" style={{width: width, height: height, backgroundColor: color}}></div>
        </div>
    )
}

export default Box