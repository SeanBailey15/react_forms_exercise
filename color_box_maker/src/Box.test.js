import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", () => {
  render(<Box width={100} height={100} color={"blue"} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Box width={100} height={100} color={"blue"} />
  );
  expect(asFragment()).toMatchSnapshot();
});
