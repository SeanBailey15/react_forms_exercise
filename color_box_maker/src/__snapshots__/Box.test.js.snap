// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <button
      class="Box-btn"
      style="display: inline-block;"
      type="submit"
    >
      X
    </button>
    <div
      class="Box-div"
      data-testid="box"
      style="width: 100px; height: 100px; background-color: blue;"
    />
  </div>
</DocumentFragment>
`;
